components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -583,20 +572,23 @@ bool JsonRpcService::SetNetwork(const std::string& chain_id,
 
 void JsonRpcService::SetNetwork(const std::string& chain_id,
                                 mojom::CoinType coin,
+                                const absl::optional<::url::Origin>& origin,
                                 SetNetworkCallback callback) {
-  if (!SetNetwork(chain_id, coin)) {
+  if (!SetNetwork(chain_id, coin, origin)) {
     std::move(callback).Run(false);
   } else {
     std::move(callback).Run(true);
   }
 }
 
 void JsonRpcService::GetNetwork(mojom::CoinType coin,
+                                const absl::optional<::url::Origin>& origin,
                                 GetNetworkCallback callback) {
-  if (!chain_ids_.contains(coin)) {
+  const auto& chain_id = GetChainId(coin, origin);
+  if (chain_id.empty()) {
     std::move(callback).Run(nullptr);
   } else {
-    std::move(callback).Run(GetChain(prefs_, chain_ids_[coin], coin));
+    std::move(callback).Run(GetChain(prefs_, chain_id, coin));
```

Comment:
How about we add below in `GetChain` and just do `std::move(callback).Run(GetChain(prefs_, chain_id, coin));` here?
```
if (chain_id.empty()) {
  return nullptr;
}
```

Reply:
fixed in 210bbec4829bea9ca6ae39e73d2bbd6bb233ad9e